jbpm-workitems/jbpm-workitems-core/src/main/java/org/jbpm/process/workitem/core/util/WidProcessor.java

```diff
@@ -119,9 +118,17 @@ public boolean postProcessWorkItemDefinition() {
             widTemplate.add("widInfo",
                             wrappedResults);
 
+            String widName = "WorkDefinitions.wid";
+            if (processingEnv.getOptions().containsKey("widName")) {
+                widName = processingEnv.getOptions().get("widName");
+            } else {
+                processingEnv.getMessager().printMessage(Diagnostic.Kind.NOTE,
+                                                         "Unable to find option \"widName\", using default.");
+            }
```

Comment:
Assuming that `getOptions()` returns a `Map`, the JDK can do this for you:

    String widName = processingEnv.getOptions().computeIfAbsent("widName", key -> {
        processingEnv.getMessager().printMessage(Diagnostic.Kind.NOTE, "Unable to find option \"widName\", using default.");
        return "WorkDefinitions.wid";
    });

Does the same thing, uses less code.

Reply:
@triceo sorry cant apply this because it returns instance of UnmodifiableMap and with that computeIfAbsent throws an UnsupportedOperationException. Will keep it in mind for next time i use this - thanks!!!